samples/msal-angular-v2-samples/angular10-browser-sample/src/app/msal/msal.broadcast.service.ts

```diff
@@ -1,25 +1,39 @@
-import { Injectable } from '@angular/core';
+import { Inject, Injectable } from '@angular/core';
 import { Observable, Subject } from 'rxjs';
-import { MsalBroadcastEvent } from './constants';
-import { AuthenticationResult, AuthError } from '@azure/msal-browser';
-
-export interface MsalBroadcastMessage {
-  type: MsalBroadcastEvent;
-  payload: AuthenticationResult | AuthError;
-}
+import { MSAL_INSTANCE } from './constants';
+import { EventType, EventMessage, EventPayload, EventError, IPublicClientApplication, InteractionType } from '@azure/msal-browser';
 
 @Injectable()
 export class MsalBroadcastService {
   private _msalSubject: Subject<any>;
   public msalSubject$: Observable<any>;
 
-  constructor() {
-    this._msalSubject = new Subject<MsalBroadcastMessage>();
+  constructor(
+    @Inject(MSAL_INSTANCE) private msalInstance: IPublicClientApplication
+  ) {
+    this._msalSubject = new Subject<EventMessage>();
     this.msalSubject$  = this._msalSubject.asObservable();
+    this.msalInstance.addEventCallback((message: EventMessage) => {
+      this._msalSubject.next(message);
+    });
```

Comment:
I think doing in the constructor is fine for now. We may want to think about moving it to its own "subscribe" or "initialize" function if there are problems (e.g. if this constructor gets invoked multiple times thus suscribing twice). Don't need to worry about for the purposes of this sample for now, but we'll want to keep keep it in mind for v2.